### DISCUSSION TITLE
eKuiper sinking back to redis encoded

### DISCUSSION THREAD

**Discussion by majidalikhan8baader** [edgexfoundry/community#104] [Q&A] [answered]
We have a scenario where we have rules in our case where we take the data from stream, process it and send it back to the redis via following rule.

```
{
  "sql": "SELECT DeviceNumber from StreamDemo GROUP BY DeviceNumber, TUMBLINGWINDOW(ss, 60)",
  "actions": [
    {
      "edgex": {
        "protocol": "redis",
        "host": "edgex-redis",
        "port": 6379,
        "topic": "MachineId60",
        "messageType": "event",
        "contentType": "application/json"
      }
    }
  ]
}
```

We can listen on the redis via command, `redis-cli subscribe MachineId60` but the modified event that we get from the redis is base64 encoded inside the payload. Is there a way to send decoded data back to the redis or am I missing something?

**Comment by ngjaying** [comment, marked as answer]
@majidalikhan8baader eKuiper EdgeX sink is used to write message back to EdgeX message bus regardless its underlying type (redis by default, but could be MQTT or NATS etc.), so the message will always be wrapped. eKuiper has a redis sink https://ekuiper.org/docs/en/latest/guide/sinks/builtin/redis.html, but it only support string or list type. You can raise a feature request in https://github.com/lf-edge/ekuiper/issues

**Reply by majidalikhan8baader** [reply]
It seems my findings were wrong, when this sink sends data back to the message bus, EdgeX decodes the encoded payload and creates an Event object with the payload as an event field and our custom application service can listen to the modified topic from the rule and work with the updated event object. 

Nonetheless, your answer helps with future questions, so thank you for reaching out.
